R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+#' @importFrom utils download.file unzip
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
+download_data_zenodo <- function(record_id, file_name, destination_folder = '', index = NULL, key = "series_name", value_column_name = "series_value"){
+  
+  if(is.null(record_id) | is.na(as.numeric(record_id)))
+    stop("Please provide the ID of the dataset that you need to download. This should be a numerical value.")
+  
+  if(is.null(file_name) | is.na(as.character(file_name)))
+    stop("Please provide the file name of the dataset that you need to download. This should be a character/string value.")
+  
+  # Check whether the destination folder exists. If not create a folder at the specified path
+  if(destination_folder != '' & !file.exists(destination_folder)){
+    tryCatch({
+      dir.create(destination_folder)
+    }, error = function(e) {   
+      message(e)
+      stop("Please provide a valid path/name for destination folder")
+    })
+  }
+    
+  if(destination_folder == ''){  
+    dest_file <- paste0(file_name, ".zip")
+    tsf_file <- paste0(file_name, ".tsf")
+  }else{  
+    dest_file <- file.path(destination_folder, paste0(file_name, ".zip"))
+    tsf_file <- file.path(destination_folder, paste0(file_name, ".tsf"))
+  }
+  
+  
+  if(!file.exists(tsf_file)){ # Check whether the required .tsf file is already there at the specified destination folder
+    if(!file.exists(dest_file)){ # Check whether the required .zip file is already there at the specified destination folder. If not download the .zip file from Zenodo
+      tryCatch({
+        download.file(paste0("https://zenodo.org/record/", record_id, "/files/", file_name, ".zip"), destfile = dest_file)
+      }, error = function(e) {   
+        message(e)
+        stop("Record ID or file name is incorrect")
+      })
+    }
+    
+    if(destination_folder != '') # Unzip the downloaded .zip folder
+      unzip(dest_file, exdir = destination_folder)
+    else
+      unzip(dest_file)
+  }
+ 
+  convert_tsf_to_tsibble(tsf_file, value_column_name, key, index) # Creating a tsibble/data.frame depending on the provided key and index
+}
+
+
+
+#' get_forecastingdata function
+#' 
+#' This function downloads a zip file from Zenodo platform given the name of the dataset, unzip the contents and converts the unzipped .tsf file into a tsibble or a data.frame given the key and index.
+#' For more details, please refer to our paper: Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @param dataset Name of the dataset that needs to be downloaded from Zenodo.
```

Comment:
I realise that the data is hosted on Zenodo, but perhaps this should say 'downloaded from the Monash Time Series Forecasting Repository'?

Reply:
Fixed as suggested